### DISCUSSION TITLE
Prompt Management: Metadata for Prompt Versions

### DISCUSSION THREAD

**Discussion by jannikmaierhoefer** [langfuse/langfuse#4304] [Ideas]
### Describe the feature or potential improvement

As prompt labels can only be applied to one prompt version (e.g. to ensure that only one version is labeled with production), it would be nice to have the option to attach metadata to a prompt version. This could be useful if teams would like to dynamically configure their application.

For instance, if we want to enable tools (rag tool), we just assign it as metadata to the prompt version.

**Comment by attila-symanto** [comment]
I would also be useful to know, which prompt settings were active at the time for a prompt version, when checking the traces.

**Comment by marcklingen** [comment]
Thanks for sharing. Have you tried the existing `config` object? In what ways would metadata work differently here?

**Reply by jannikmaierhoefer** [reply]
"There are non-technical users managing prompts, and it is a bit difficult to explain to them to create a valid JSON"

**Reply by marcklingen** [reply]
Thanks for the context, I don't yet understand how this would this be different for a `metadata` object?
